back/engines/commercial/user_custom_fields/app/models/user_custom_fields/representativeness/r_score.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+class UserCustomFields::Representativeness::RScore
+  attr_reader :value, :user_counts, :ref_distribution
+
+  # @param [Numeric] value
+  # @param [UserCustomFields::Representativeness::RefDistribution] ref_distribution
+  def initialize(value, user_counts, ref_distribution)
+    @value = value
+    @user_counts = user_counts
+    @ref_distribution = ref_distribution
+    @timestamp = Time.zone.now
+  end
+
+  def id
+    "#{@ref_distribution.id}_#{@timestamp.to_i}_rscore"
+  end
+
+  class << self
+    # Factory method that instantiates an RScore based on observed user counts and a reference distribution.
+    # @param [Hash] user_counts
+    # @param [UserCustomFields::Representativeness::RefDistribution] ref_distribution
+    # @return [UserCustomFields::Representativeness::RScore]
+    def compute(user_counts, ref_distribution)
+      population_counts = ref_distribution.distribution
+      check_user_counts_options!(user_counts, ref_distribution)
+
+      score_value = compute_scores(user_counts, population_counts)[:min_max_p_ratio]
+      new(score_value, user_counts, ref_distribution)
+    end
+
+    # Compute the ratio between the minimum and maximum participation rates.
+    # @param [Hash] user_counts
+    # @param [Hash] population_counts
+    # @return [Numeric]
+    def min_max_p_ratio(user_counts, population_counts)
+      min_rate, max_rate = participation_rates(user_counts, population_counts).minmax
+      min_rate / max_rate
+    end
+
+    # Compute the ratio between the minimum and mean participation rates.
+    # @param [Hash] user_counts
+    # @param [Hash] population_counts
+    # @return [Numeric]
+    def min_mean_p_ratio(user_counts, population_counts)
+      participation_rates = send(:participation_rates, user_counts, population_counts)
+      min_rate = participation_rates.min
+      mean_rate = participation_rates.sum.to_f / participation_rates.size
+      min_rate / mean_rate
+    end
+
+    def participation_rates(user_counts, population_counts)
+      population_counts.keys.map do |option_id|
```

Comment:
Related to my comment below regarding `check_user_counts_options!`:

If we use `population_counts.keys` to `.map` over, and I assume that the user distribution can have keys that are not in the reference distribution (because of the admin choosing to disable fields, which would currently throw an error in `check_user_counts_options!`), we can get the following situation: 

All the fields that the admin has selected happen to have a user count of zero. In this case, all `participation_rates` are zero. If you then calculate the `min_max_p_ratio` or the `min_mean_p_ratio` you are dividing by zero. Maybe there should be an extra check to make sure this is not the case? I was thinking of adding this check in the FE as well just for the population data because if the absolute population values are all 0 the percentage calculation would get messed up...

Reply:
By the way, might be more related to the previous PR, but how are you calculating the `expected_users` exactly? Because I think we should only calculate the expected users based on the keys that are actually in the reference distribution, which I think is not the case right now. In the question I asked you yesterday about the `expected_users` (see image) the expected user counts are ` { male: 3.8, female: 4.2 }` even though the absolute numbers of users in the dashboard are ` { male: 1, female: 1 }` (because the reference distribution has no value for `unspecified`). Shouldn't the sum of expected user counts be equal to the total user count, i.e. ignore `unspecified` when it's calculated?

Sorry really choatic story haha, hope it makes sense!

![image](https://user-images.githubusercontent.com/13822005/175038135-1577eda1-e222-4558-9759-5e9a2b93a402.png)